qa/tasks/cephfs/test_cephfs_shell.py

```diff
@@ -1,17 +1,41 @@
 import os
 import crypt
 import logging
+from tempfile import mkstemp as tempfile_mkstemp
 from StringIO import StringIO
 from tasks.cephfs.cephfs_test_case import CephFSTestCase
 from tasks.cephfs.fuse_mount import FuseMount
 from teuthology.exceptions import CommandFailedError
 
 log = logging.getLogger(__name__)
 
+# XXX: mount_x needs to be defined in setup and destroyed/defined to None
+# teardown of every class.
+mount_x = None
+
+def run_cephfs_shell_script(script):
+    scriptpath = tempfile_mkstemp(prefix='test-cephfs', text=True)[1]
```

Comment:
This tmpfile is created on the machine running the test which probably is not the same machine running `cephfs-shell`. You should look at `remote.put_file` as a way to copy the script over.

Reply:
Ok. I've made the change.

vstart_runner.py's `put_file` had issues since source and destination locations would be same for it. I've made it ignore the exception raised when have source and destination are same for `shutil.copy`